### PR TITLE
refactor: bitcoin send flow to accept multiple recipients

### PR DESCRIPTION
**Pull request by alter-eggo**
> Try out Leather build a89dbca — [Extension build](https://github.com/leather-wallet/extension/actions/runs/8925771572), [Test report](https://leather-wallet.github.io/playwright-reports/refactor/multiple-recipients), [Storybook](https://refactor-multiple-recipients--65982789c7e2278518f189e7.chromatic.com), [Chromatic](https://www.chromatic.com/library?appId=65982789c7e2278518f189e7&branch=refactor/multiple-recipients)

This pr refactors bitcoin send flows to accept multiple recipients, like it is in rpc send transfer flow

### CODE REVIEW

**Comment by markmhendrickson** [comment]
What are the "Bitcoin send flows" in particular – are these separate API(s) from "send transfer flows"?

**Comment by alter-eggo** [comment]
@markmhendrickson 
- btc send flow
- brc-20 send flow
- inscription send flow
- rpc send transfer flow

to support multiple recipients feature, I had to duplicate a lot of code. in this pr I removed it

File: src/app/common/transactions/bitcoin/coinselect/local-coin-selection.ts [hunk 1]
```diff
@@ -168,62 +57,67 @@ export function determineUtxosForSpendAllMultipleRecipients({
...
-  const filteredUtxos = filterUneconomicalUtxosMultipleRecipients({
-    utxos: orderedUtxos,
+  const filteredUtxos = filterUneconomicalUtxos({
+    utxos: utxos.sort((a, b) => b.value - a.value),
     feeRate,
     recipients,
   });
+  if (!filteredUtxos.length) throw new InsufficientFundsError();
+
+  const amount = recipients.reduce((acc, { amount }) => acc + amount.amount.toNumber(), 0);
```
Comment (kyranjamie) [inline src/app/common/transactions/bitcoin/coinselect/local-coin-selection.ts]:
```suggestion
  const amount = sumMoney(recipients.map(recipient => recipient.amount));
```

File: src/shared/models/form.model.ts [hunk 2]
```diff
@@ -37,3 +38,8 @@ export interface StacksTransactionFormValues {
   feeType: string;
   nonce?: number | string;
 }
+
+export interface BitcoinRecipient {
```
Comment (kyranjamie) [inline src/shared/models/form.model.ts]: There's nothing bitcoin specific about this. How could we name differently?

`TransferRecipient`?

**Review by kyranjamie — APPROVED** [review]
Great work @alter-eggo, love to see this 
image

File: src/app/common/transactions/bitcoin/coinselect/local-coin-selection.ts (hunk 1 above)
Comment (kyranjamie) [reply src/app/common/transactions/bitcoin/coinselect/local-coin-selection.ts]: reduce is great but it isn't super readable, so use helpers where possible

**Pull request merged**